Scripts/RemediationScripts/Remediate-EnableHttpsForStorageAccount.ps1

```diff
@@ -0,0 +1,723 @@
+    #>  
+    
+    Write-Host "Required modules are: Az.Account, Az.Resources, Az.Storage" -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host "Checking for required modules..."
+    $availableModules = $(Get-Module -ListAvailable Az.Resources, Az.Accounts,Az.Storage)
+    
+    # Checking if 'Az.Accounts' module is available or not.
+    if($availableModules.Name -notcontains 'Az.Accounts')
+    {
+        Write-Host "Installing module Az.Accounts..." -ForegroundColor $([Constants]::MessageType.Info)
+        Install-Module -Name Az.Accounts -Scope CurrentUser -Repository 'PSGallery'
+        Write-Host " Module Az.Accounts is installed." -ForegroundColor $([Constants]::MessageType.update)
+    }
+    else
+    {
+        Write-Host "Az.Accounts module is available." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    
+    # Checking if 'Az.Storage' module with required version is available or not.
+    if($availableModules.Name -notcontains 'Az.Storage')
+    {
+        Write-Host "Installing module Az.Storage..." -ForegroundColor $([Constants]::MessageType.Info)
+        Install-Module -Name Az.Storage -Scope CurrentUser -Repository 'PSGallery'
+        Write-Host "Module Az.Storage is installed." -ForegroundColor $([Constants]::MessageType.update)
+    }
+    else
+    {
+        Write-Host "Az.Storage module is available." -ForegroundColor $([Constants]::MessageType.Update)
+        $currentModule = $availableModules | Where-Object { $_.Name -eq "Az.Storage" }
+        $currentModuleVersion = ($currentModule.Version  | measure -Maximum).Maximum -as [string]
+       
+        if([version]('{0}.{1}.{2}' -f $currentModuleVersion.split('.')) -lt [version]('{0}.{1}.{2}' -f "3.5.0".split('.')))
+        {
+            Write-Host "Updating module Az.Storage..." -ForegroundColor Blue $([Constants]::MessageType.Info)
+            Update-Module -Name "Az.Storage"
+            Write-Host " Module Az.Storage is updated." -ForegroundColor $([Constants]::MessageType.update)
+        }
+    }
+
+    # Checking if 'Az.Resources' module is available or not.
+    if($availableModules.Name -notcontains 'Az.Resources')
+    {
+        Write-Host "Installing module Az.Resources..." -ForegroundColor $([Constants]::MessageType.Info)
+        Install-Module -Name Az.Resources -Scope CurrentUser -Repository 'PSGallery'
+        Write-Host "Module Az.resources is installed..." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+    else
+    {
+        Write-Host "Az.Resources module is available." -ForegroundColor $([Constants]::MessageType.Update)
+    }
+}
+
+
+
+
+
+function Enable-StorageEncryptionInTransit
+{
+    
+     <#
+        .SYNOPSIS
+        This command would help in remediating "Azure_Storage_DP_Encrypt_In_Transit" control.
+
+        .DESCRIPTION
+        This command would help in remediating  "Azure_Storage_DP_Encrypt_In_Transit" control.
+        Enables HTTPS on the Storage Account in the Subscription. 
```

Comment:
While "Enables HTTPS" sounds okay, would it be more appropriate to have "Enable secure transfer..." instead?

Reply:
yes agree.